app/routes/__index/certificate/index.tsx

```diff
@@ -1,57 +1,80 @@
 import { Flex, Center } from '@chakra-ui/react';
-import type { Certificate } from '@prisma/client';
-import { useState } from 'react';
+import type { LoaderArgs, ActionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import { useTypedLoaderData } from 'remix-typedjson';
+import { useRevalidator } from '@remix-run/react';
+import { useInterval } from 'react-use';
 
-import undrawSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import { requireUser } from '~/session.server';
+import pendingSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import failedSvg from '~/assets/undraw_cancel_re_pkdm.svg';
 import Loading from '~/components/display-page';
 import CertificateAvailable from '~/components/certificate/certificate-available';
 import CertificateRequestView from '~/components/certificate/certificate-request';
 import { useUser } from '~/utils';
+import { getCertificateStatusByUsername } from '~/models/certificate.server';
+import { addCertRequest } from '~/queues/certificate/certificate-flow.server';
 
-export default function CertificateIndexRoute() {
-  const user = useUser();
-  const [certificateRequested, setCertificateRequested] = useState(false);
-  const [loading, setLoading] = useState(false);
+export const loader = async ({ request }: LoaderArgs) => {
+  const user = await requireUser(request);
 
-  const currentDate: Date = new Date(2023, 2);
+  return await getCertificateStatusByUsername(user.username);
+};
 
-  const certificate: Certificate = {
-    id: 1,
+export const action = async ({ request }: ActionArgs) => {
+  const user = await requireUser(request);
+
+  await addCertRequest({
```

Comment:
I would confirm the request verb here (e.g., `POST`?)

Reply:
Also, you should probably confirm that there isn't a request already pending.  I know you disable the UI to do this, but someone could `POST` manually to do this again.

Reply:
Oh yeah I see the problem thank you I will change it. Also, by "confirming" what do you mean exactly?

Reply:
Check what the request method is